src/pcap-helper.c

```diff
@@ -0,0 +1,75 @@
+/* Copyright (C) 2007-2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "pcap-helper.h"
+
+void SplitPcapDump(u_char *user, struct pcap_pkthdr *libpcap_hdr,
+        const u_char *p1, bpf_u_int32 p1_len, const u_char *p2,
+        bpf_u_int32 p2_len) {
+    /*
+     * Check for overflow due to packet lengths.
+     */
+    if ((p1_len + p2_len) < p1_len) {
+        fprintf(stderr,"Error in SplitPcapDump. Overflow due to packet "
```

Comment:
`SCLogError` or warning?

Reply:
I'll switch these to SCLogError. Initially I just wrote this in the same style as libpcap's dump_pcap function. I think I'll try and get this SplitPcapDump function added to their pcap API. For now though, I'll rewrite these pcap-helper files in the same code style as Suricata.